File: 03-Stack/suhee0206/B-10828.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int N;
vector<int> stack;
string x;
int top = -1;

void command(string x, vector<int> &stack) {
	if (x == "push") {
		int num;
		cin >> num;
		top++;
		stack.push_back(num);
	}
	else if (x == "pop") {
		if (stack.empty()) {
			cout << -1 << '\n';
		}
		else {
			cout << stack[top] << '\n';
			stack.pop_back();
			top--;
		}
	}
	else if (x == "size") {
		cout << stack.size() << '\n';
	}
	else if (x == "empty") {
		if (stack.empty()) {
			cout << 1 << '\n';
		}
		else {
			cout << 0 << '\n';
		}
	}
	else if (x == "top") {
		if (stack.empty()) {
			cout << -1 << '\n';
		}
		else {
			cout << stack[top] << '\n';
		}
	}
}

int main() {
	cin >> N;

	for (int i = 0; i < N; i++) {
		cin >> x;
		command(x, stack);
	}
}

File: 03-Stack/sds111888/B-10828.py
import sys

input=sys.stdin.readline
n = int(input())

stack = []
point = -1

#command = ["push","pop","size","empty","top"]

for i in range(n):
    c = list(input().split())
    
    if c[0] == "push":
        c[1] = int(c[1])
        stack.append(c[1])
        point += 1
    elif c[0] == "pop":
        if point >= 0:
            print(stack[point])
            del stack[point]
            point -= 1
        else :
            print(-1)
    elif c[0] == "size":
        print((point+1))
    elif c[0] == "empty":
        if point == -1 :
            print(1)
        else :
            print(0)
    elif c[0] == "top":
        if point >= 0:
            print(stack[point])
        else :
            print(-1)